prims/prims.c

```diff
@@ -0,0 +1,47 @@
+#include <stdio.h>
+#include <string.h>
+
+#define SIZE 5
+#define INFINITY 9999
+
+// This function finds the minimal spanning tree by Prim's Algorithm
+void prims(int G[SIZE][SIZE], int *parent) {
+    int select[SIZE], i, j, k;
+    int v1 = 0, v2 = 0;
+    for (i = 0; i < SIZE; ++i)  // Initialize the selected vertices list
+        select[i] = 0;
+    select[0] = 1;
+    for (k = 1; k < SIZE; ++k) {
+        int min_dist = INFINITY;
+        for (i = 0; i < SIZE; ++i) {  // Select an edge such that one vertex is selected and other is not and the edge
+            for (j = 0; j < SIZE; ++j) {  // has the least weight.
+                if (G[i][j] && ((select[i] && !select[j]) || (!select[i] && select[j]))) {
+                    if (G[i][j] < min_dist) {  //obtained edge with minimum wt 
+                        min_dist = G[i][j];
+                        v1 = i;
+                        parent[j] = v1;
+                        v2 = j;  //picking up those vertices
+                    }
+                }
+            }
+        }
+        select[v1] = select[v2] = 0; 
```

Comment:
If we do ` select[v1] = select[v2] = 0;` your code throws RE(runtime error), [click here](http://ideone.com/BzBGDD)
And remove extra whitespace at the end of this line(28).